core/windows_and_tabs/window_snap.py

```diff
@@ -302,14 +328,170 @@ def __init__(self, left, top, right, bottom):
 }
 
 
+def _snap_next(windows: list[Any], target_layout: RelativeScreenPos) -> Optional[int]:
+    print(windows)
+    for index, window in enumerate(windows):
+        if window is None:
+            return index
+        try:
+            _snap_window_helper(
+                window,
+                target_layout,
+            )
+            return index
+        except Exception as e:
+            print(f"Failed to snap window: {e}")
+    return None
+
+
+def _snap_layout(window_layout: WindowLayout):
+    """Split the screen between multiple windows."""
+    global last_focus_was_done_by_snap_layout
+    global currently_snapping_layout
+    currently_snapping_layout = True
+    import copy
+
+    target_layout = copy.deepcopy(window_layout.layout)
+    remaining_windows = window_layout.windows
+    snapped_windows = []
+    snapped_window_index = 0
+    if window_layout.should_rotate:
+        print("Trying to rotate")
+        target_layout.append(target_layout.pop(0))
+
+    while len(target_layout) > 0:
+        snapped_window_index = _snap_next(remaining_windows, target_layout.pop(0))
+        if snapped_window_index is not None:
+            snapped_windows.insert(0, remaining_windows[snapped_window_index])
+            remaining_windows = remaining_windows[snapped_window_index + 1 :]
+
+    if window_layout.should_rotate and len(snapped_windows) > 0:
+        snapped_windows.append(snapped_windows.pop(0))
+    for window in snapped_windows:
+        window.focus()
+        actions.sleep("180ms")
+    actions.sleep("500ms")
+    last_focus_was_done_by_snap_layout = True
+    currently_snapping_layout = False
+
+
+def _top_n_windows() -> list[Any]:
+    active_window = ui.active_window()
+    ui_windows = ui.windows()
+    try:
+        active_index = ui_windows.index(active_window)
+        return ui_windows[active_index:]
+    except Exception:
+        return ui_windows
+
+
 @mod.capture(rule="{user.window_snap_positions}")
 def window_snap_position(m) -> RelativeScreenPos:
     return _snap_positions[m.window_snap_positions]
 
 
-@mod.capture(rule="{user.window_split_positions}")
-def window_split_position(m) -> Dict[int, list[RelativeScreenPos]]:
-    return _split_positions[m.window_split_positions]
+@mod.capture(rule="{user.window_split_positions} [<number_small>]")
+def window_split_position(m) -> list[RelativeScreenPos]:
+    if hasattr(m, "number_small"):
+        return _split_positions[m.window_split_positions][m.number_small]
+    else:
+        min_key = min(_split_positions[m.window_split_positions].keys())
+        return _split_positions[m.window_split_positions][min_key]
+
+
+@mod.capture(rule="all")
+def all_windows(m) -> list[Any]:
```

Comment:
This also seems a strange name for the capture as it does not return all windows.

Reply:
I renamed it to reflect the fact that the windows are under the active window but we don't know whether they are resizable or not-there is not really a good way to detect that windows can be resized other than to try and resize them so some of these windows will be skipped.